src/audio/codec_adapter/codec/generic.c

```diff
@@ -182,8 +182,7 @@ int codec_free_memory(struct comp_dev *dev, void *ptr)
 	struct list_item *_mem_list;
 
 	if (!ptr) {
```

Comment:
Maybe we should make return type from codec_free_memory void. Isn't that we can recover or something is this fails.

Reply:
I was thinking about this, but wasn't sure if we wanted to lose the information if we tried to free something outside the container

Reply:
I think this is good to merge now, we can look at the return types as part of codec adapter work.